src/it/scala/tech/cryptonomic/conseil/client/EndpointClient.scala

```diff
@@ -0,0 +1,169 @@
+package tech.cryptonomic.conseil.client
+
+import java.util.concurrent.Executors
+import scala.language.postfixOps
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.sys.process._
+import scala.Console.{GREEN, RED, RESET}
+import org.http4s.client.blaze._
+import org.http4s.Header
+import org.http4s.headers._
+import org.http4s.MediaType
+import org.http4s.circe._
+import org.http4s.dsl.io._
+import org.http4s.client.dsl.io._
+import org.http4s.Uri
+import io.circe.{Json, parser}
+import cats.effect.IO
+import cats.syntax.apply._
+
+/** Currently can be used to test any conseil instance that loaded blocks levels 1 to 1000
+  * against predefined expectations on the responses
+  */
+object DataEndpointsClientProbe {
+
+  object Setup {
+    implicit val ioShift = IO.contextShift(scala.concurrent.ExecutionContext.global)
+    val timer = IO.timer(scala.concurrent.ExecutionContext.global)
+
+    /* We might wanna start with only 10k blocks */
+    private def runLorre(network: String) =
+      Seq("sbt", s"runLorre -d 10000 -h BLc7tKfzia9hnaY1YTMS6RkDniQBoApM4EjKFRLucsuHbiy3eqt $network")
+    private val runConseil = Process(Seq("sbt", "runConseil -v"))
+    private def grepConseilPID(): String = (Seq("jps", "-m") #| Seq("grep", "runConseil")).!!.split(" ").head
+    private def stopPid(pid: String) = Seq("kill", "-2", pid).!
+
+    private def syncData(network: String) = IO((runLorre(network).!).ensuring(_ == 0, "lorre failed to load correctly"))
+
+    private def startConseil(syncNetwork: Option[String]): IO[Process] =
+      for {
+        _ <- if (syncNetwork.nonEmpty) syncData(syncNetwork.get) else IO(0)
+        proc <- IO(runConseil.run())
+        _ <- IO(println("waiting for conseil to start"))
+        _ <- timer.sleep(30.seconds)
+      } yield proc
+
+    def usingConseil[A](syncNetwork: Option[String] = None)(testBlock: => IO[A]) =
+      startConseil(syncNetwork).bracket(
+        use = _ => testBlock
+      )(
+        release = _ =>
+          for {
+            pid <- IO(grepConseilPID())
```

Comment:
why wrapping each line in `IO`?

Reply:
For consistency of execution? Otherwise side-effects might come out intertwined incorrectly. It's essentially sequencing explicitly the operations.